tracer/test/test-applications/integrations/Samples.DataStreams.ManualAPI/Program.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace;
+
+namespace Samples.DataStreams.ManualAPI;
+
+public class Program
+{
+    private static ConcurrentQueue<string> _queue = new();
+
+    private static async Task Main(string[] args)
+    {
+        var writing = Send("my message");
+        var reading = Receive();
+
+        // wait up to 20 second for threads
+        await Task.WhenAny(Task.WhenAll(reading, writing), Task.Delay(TimeSpan.FromSeconds(value: 20)));
+    }
+
+    private static async Task Send(string message)
+    {
+        using var scope = Tracer.Instance.StartActive("Samples.DataStreams.ManualAPI.Send");
+
+        await Task.Delay(millisecondsDelay: 100);
+
+        Console.WriteLine("Sending one message to the queue...");
+        var sb = new StringBuilder();
+        var injector = new SpanContextInjector();
+        injector.InjectIncludingDsm(sb, (b, k, v) => b.Append($"{k}:{v};"), scope.Span.Context, "ConcurrentQueue", nameof(_queue));
+        sb.Append(message);
+        _queue.Enqueue(sb.ToString());
+        Console.WriteLine("message sent");
+    }
+
+    private static async Task<string> Receive()
+    {
+        Console.WriteLine("Receiving one message from the queue...");
+        string result;
+        while (!_queue.TryDequeue(out result))
+        {
+            await Task.Delay(millisecondsDelay: 100);
+            Console.WriteLine("Retrying to receive");
+        }
+
+        var headers = Parse(result, out var content);
+        Console.WriteLine($"Parsed {headers.Count} headers");
+        var extractor = new SpanContextExtractor();
+        var extractedContext = extractor.ExtractIncludingDsm(
+            headers,
+            // complicated getter because we need to return an empty array if no result
+            (d, k) =>
+            {
+                if (d.TryGetValue(k, out var val))
+                {
+                    return [val];
+                }
+
+                return [];
+            },
```

Comment:
Seeing how this is used, what would you think about adding an override to `ExtractIncludingDsm` where instead of a getter function we provide just the header key and the getter is built inside to simplify usage?

Reply:
I don't think that's possible, we need a getter function to know _how_ to extract the header 🤔 We can't easily do it automatically, because we can't add references to the message types (for example). We could add a whole serious of `TryDuckCast<>` to try to "figure out" how to extract the value, but that would be very inefficient...

Reply:
what we _could_ do to make it easier to use is to consider `null` as "key not found". I don't know if it would have deeper implications with the existing code.

Reply:
but actually... even that wouldn't change much I think. It'd be nice to write the getter as `(d, k) => [d[k]]` but then we'd have to handle `KeyNotFoundException` as missing key, and I don't think we want to do that.